plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/detailviews/AbstractStackedDetailViews.java

```diff
@@ -165,6 +171,56 @@ public Composite createControls(Composite parent, Object context, DataBindingCon
 		public boolean isViewFor(Object object) {
 			return false;
 		}
+
+		/**
+		 * Helper method that creates Label + read-only StyledText
+		 * @param labelText
+		 * @param container
+		 * @return
+		 */
+		protected StyledText createLabeledValue(String labelText, Composite container) {
+			Label label = new Label(container, SWT.NONE);
+			label.setText(labelText);
+			GridDataFactory.fillDefaults()
+				.align(SWT.LEFT, SWT.CENTER).applyTo(label);
+			return createValue(container);
+		}
+
+		/**
+		 * Helper method that creates read-only StyledText for a value to display
+		 * @param container
+		 * @return
+		 */
+		protected StyledText createValue(Composite container) {
+			StyledText styledText = new StyledText(container, SWT.READ_ONLY);
+			styledText.setAlwaysShowScrollBars(false);
+			StyledTextUtils.setTransparent(styledText);
+			GridDataFactory.fillDefaults()
+					.align(SWT.LEFT, SWT.CENTER).grab(true, false).applyTo(styledText);
+			styledText.addFocusListener(fl);
+			return styledText;
+		}
+
+		/**
+		 * Listener used to remove selection from read-only text widget when focus is lost.
+		 * This prevents independent selecting on several widgets. As selection is used
+		 * for copying text to the clipboard, it is not needed without focus. 
+		 */
+		protected FocusListener fl = new FocusListener() {
+
+			@Override
+			public void focusGained(FocusEvent e) {
+			}
+
+			@Override
+			public void focusLost(FocusEvent e) {
+				if(e.getSource() instanceof StyledText) {
```

Comment:
Unfortunately, setSelection() is not an interface or superclass method. It is defined on StyledText, Text separately. Ok, I added case of Text even though it is not used now.